### PR TITLE
Seal the compiler

### PR DESCRIPTION
**Pull request by ike709**
Imports Robust's analyzers, mainly the one to mark classes as `sealed` whenever possible as a really minor (but free) optimization.

### CODE REVIEW

File: DMCompiler/Compiler/DM/DMASTHelper.cs [hunk 1]
```diff
@@ -7,11 +7,12 @@
 using OpenDreamShared.Dream;
 using OpenDreamShared.Dream.Procs;
 using OpenDreamShared.Compiler;
+using Robust.Shared.Analyzers;
```
Comment (wixoaGit) [inline DMCompiler/Compiler/DM/DMASTHelper.cs:10]: This should be using the VirtualAttribute from `DMCompiler` not `Robust.Shared.Analyzers`

**Review by wixoaGit — CHANGES_REQUESTED** [review]

**Comment by ike709** [comment]
I nuked the analyzer since it relies on the Robust namespace. This just seals up everything that the analyzer warned about.

File: DMCompiler/Compiler/DM/DMParser.cs [hunk 2]
```diff
@@ -5,8 +5,8 @@
 using OpenDreamShared.Compiler;
 using OpenDreamShared.Dream;
 using OpenDreamShared.Dream.Procs;
+using Robust.Shared.Analyzers;
```
Comment (wixoaGit) [inline DMCompiler/Compiler/DM/DMParser.cs:8]:
```suggestion
```

File: DMCompiler/DMCompiler.csproj [hunk 3]
```diff
@@ -15,6 +15,7 @@
   <ItemGroup>
     <ProjectReference Include="..\OpenDreamShared\OpenDreamShared.csproj" />
   </ItemGroup>
+
```
Comment (wixoaGit) [inline DMCompiler/DMCompiler.csproj:18]:
```suggestion
```

**Review by wixoaGit — CHANGES_REQUESTED** [review]

**Pull request merged**